frontend/src/metabase/common/components/FilterPicker/FilterPicker.tsx

```diff
@@ -121,13 +124,18 @@ export function FilterPicker({
   const FilterWidget = getFilterWidget(column);
 
   if (FilterWidget) {
+    const isSameColumn =
+      initialColumn &&
+      Lib.isSameColumn(query, stageIndex, initialColumn, column);
+
     return (
       <Box miw={MIN_WIDTH}>
         <FilterWidget
           query={query}
           stageIndex={stageIndex}
           column={column}
-          filter={filter}
+          filter={isSameColumn ? filter : undefined}
+          isNew={!filter}
```

Comment:
I don't love that picker components can't derive the `isNew` state from `filter` any longer, but this seems like a reasonable compromise?

Reply:
LGTM